### PR TITLE
Feature/admin page

### PR DESCRIPTION
**Pull request by Khushboo1008**

### CODE REVIEW

File: src/containers/admin/components/ConfirmButton.tsx [hunk 1]
```diff
@@ -89,18 +107,47 @@ const ConfirmButton = () => {
...
+        console.log(size)
+        if (size > 25) {
+          let response
+          const ActualSize = Math.floor(size / 25)
+          console.log(ActualSize)
+          let chunk
+          for (let i = 0; i <= ActualSize; i++) {
+            if (i === ActualSize && size % 25 !== 0) {
+              chunk = batch.inputParams.slice(i * 25, size)
+              response = await addChunk(chunk, nextBatchId)
```
Comment (codebuster22) [inline src/containers/admin/components/ConfirmButton.tsx:121]: what if the response is 201 here, that means Email wasn't sent but still you are sending more

File: src/containers/admin/components/ConfirmButton.tsx [hunk 2]
```diff
@@ -89,18 +107,47 @@ const ConfirmButton = () => {
...
+              setMailSent(size)
+            } else if (i !== ActualSize) {
+              chunk = batch.inputParams.slice(i * 25, (i + 1) * 25)
+              response = await addChunk(chunk, nextBatchId)
```
Comment (codebuster22) [inline src/containers/admin/components/ConfirmButton.tsx:125]: same

**Review by codebuster22 — CHANGES_REQUESTED** [review]

File: src/containers/admin/components/ConfirmButton.tsx [hunk 3]
```diff
@@ -34,6 +35,8 @@ const ConfirmButton = () => {
   const batch = useAppSelector(batchSelector)
   const dispatch = useAppDispatch()
   const [allowed, setAllowed] = useState<boolean>(false)
+  const [mailSent, setMailSent] = useState<number>(0)
+  const MAX_SIZE = 25
```
Comment (codebuster22) [inline src/containers/admin/components/ConfirmButton.tsx:39]: What max size are we talking about. Either add comment or make variable more descriptive

File: src/containers/admin/components/ConfirmButton.tsx [hunk 4]
```diff
@@ -79,6 +82,23 @@ const ConfirmButton = () => {
     await addBatchToContract(merkleRoot, cid, nextBatchId)
   }
 
+  const addChunk = async (chunk, nextBatchId) => {
```
Comment (codebuster22) [inline src/containers/admin/components/ConfirmButton.tsx:85]: Add comments on what this function does

File: src/containers/admin/components/ConfirmButton.tsx [hunk 5]
```diff
@@ -89,18 +109,53 @@ const ConfirmButton = () => {
...
-          batchId: nextBatchId.toString(),
-          eventName: 'Vivacity 2023',
-          contractAddress: CONTRACT_ADDRESS,
-          addBatchTimestamp: Date.now(),
-        }
-        const response = await sendDataToServer(serverData)
-        if (response.status !== 200) {
-          toast(`❌ Something went wrong! Please Try Again`)
+        setLoading(true)
+        const size = batch.inputParams.length
```
Comment (codebuster22) [inline src/containers/admin/components/ConfirmButton.tsx:113]: What does size represent. Make variable descriptive

File: src/containers/admin/components/ConfirmButton.tsx [hunk 6]
```diff
@@ -89,18 +109,53 @@ const ConfirmButton = () => {
...
+        console.log(size)
+        if (size > MAX_SIZE) {
+          let response
```
Comment (codebuster22) [inline src/containers/admin/components/ConfirmButton.tsx:116]: What response is this. Make it more descriptive

File: src/containers/admin/components/ConfirmButton.tsx [hunk 7]
```diff
@@ -89,18 +109,53 @@ const ConfirmButton = () => {
...
+          const ActualSize = Math.floor(size / MAX_SIZE)
```
Comment (codebuster22) [inline src/containers/admin/components/ConfirmButton.tsx:117]: What does actual size represent? Make it more descriptive

**Review by codebuster22 — CHANGES_REQUESTED** [review]
Make variables more descriptive. Cannot merge if variables are not understandable.

**Review by codebuster22 — APPROVED** [review]
Skimmed it, but it looks better now

**Review by angel112 — APPROVED** [review]

**Pull request merged**